backend/src/ee/services/dynamic-secret-lease/dynamic-secret-lease-dal.ts

```diff
@@ -12,7 +12,10 @@ export const dynamicSecretLeaseDALFactory = (db: TDbClient) => {
 
   const countLeasesForDynamicSecret = async (dynamicSecretId: string, tx?: Knex) => {
     try {
-      const doc = await (tx || db)(TableName.DynamicSecretLease).count("*").where({ dynamicSecretId }).first();
+      const doc = await (tx || db.replicaNode())(TableName.DynamicSecretLease)
```

Comment:
hmm, there is no better way to somehow do this dynamically without adding db.replicaNode() all through out the code base?

Reply:
Is this what drizzle does in their code base? I know there is a on going talk about it here on knex js https://github.com/knex/knex/issues/2253#issuecomment-551610832

Reply:
Dynamically not possible as knexjs itself doesn't allow anything intermediate conversion. 

I have seen the above thread and many more. The biggest issue with all the issues is first its not reliable. People are reporting x or y not working. Transactions breaking etc. Second overriding something internal knexjs could also cause overhead maintenance in future if they change the structure. 

The `replicaNode()` need to be made only on read query that too custom written and not using orm.

Reply:
https://github.com/drizzle-team/drizzle-orm/blob/4ecfe1fca3ec0bc8fdec02b6751e030b8832fb41/drizzle-orm/src/pg-core/db.ts#L619